### ISSUE TITLE
POST request

### ISSUE THREAD

**Issue by meeximum** [Unleash/unleash-proxy#37]
Hi,

I saw that it's possible to send a POST request with context information to the proxy, but when I try with Postman, I got an error 

`<pre>TypeError: Cannot read properties of undefined (reading &#39;map&#39;)<br> &nbsp; &nbsp;at Client.getDefinedToggles (C:\private\git\node_modules\@unleash\proxy\dist\client.js:75:28)<br> &nbsp; &nbsp;at UnleashProxy.lookupToggles (C:\private\git\node_modules\@unleash\proxy\dist\unleash-proxy.js:61:41)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (C:\private\git\node_modules\express\lib\router\layer.js:95:5)<br> &nbsp; &nbsp;at next (C:\private\git\node_modules\express\lib\router\route.js:137:13)<br> &nbsp; &nbsp;at Route.dispatch (C:\private\git\node_modules\express\lib\router\route.js:112:3)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (C:\private\git\node_modules\express\lib\router\layer.js:95:5)<br> &nbsp; &nbsp;at C:\private\git\node_modules\express\lib\router\index.js:281:22<br> &nbsp; &nbsp;at Function.process_params (C:\private\git\node_modules\express\lib\router\index.js:335:12)<br> &nbsp; &nbsp;at next (C:\private\git\node_modules\express\lib\router\index.js:275:10)<br> &nbsp; &nbsp;at Function.handle (C:\private\git\node_modules\express\lib\router\index.js:174:3)</pre>`

My Payload
`{"userId":"test"}`

Best regards
Meex

**Comment by ivarconr** [comment]
Thanks for reaching out, and sorry for a bit late response, due to holidays. 

The POST endpoint is not very documented as it was mainly build as a proof of concept. 

The payload is required to take both context and toggle names you want to evaluate, like this:
{"context": {"userId": "123"}, "toggles": ["demo"]}

Example with HTTPie:

```sh
echo '{"context": {"userId": "123"}, "toggles": ["demo"]}' | http POST http://localhost:3000/proxy Authorization:s1
HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: ETag
Connection: keep-alive
Content-Length: 79
Content-Type: application/json; charset=utf-8
Date: Thu, 06 Jan 2022 19:25:20 GMT
ETag: W/"4f-C2Uz6jU0DkutatGC8SCJOy5to7A"
Keep-Alive: timeout=5
Vary: Accept-Encoding

[
    {
        "enabled": false,
        "name": "demo",
        "variant": {
            "enabled": false,
            "name": "disabled"
        }
    }
]

```

The proxy should however not crash for an invalid payload. Will add a fix for that!

**Comment by ivarconr** [comment]
fixed in 5c7ce019903f8f90423874e5a366e384713d8263
